### PR TITLE
(feat) Show asterisks next to labels for required fields

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR adds a red asterisk to form labels for required fields as shown below:

## Screenshots

https://github.com/openmrs/openmrs-form-engine-lib/assets/8509731/2cda3b00-520a-4c31-ad33-f3d2b0afb124

### CODE REVIEW

File: src/form-engine.test.tsx [hunk 1]
```diff
@@ -116,12 +109,14 @@ describe('Form component', () => {
...
     await assertFormHasAllFields(screen, [
-      { fieldName: 'When was the HIV test conducted?', fieldType: 'date' },
+      { fieldName: 'When was the HIV test conducted? *', fieldType: 'date' },
       { fieldName: 'TB screening', fieldType: 'combobox' },
     ]);
 
     try {
       await findSelectInput(screen, 'Community service delivery point');
+      // await findSelectInput(screen, 'Community service');
+      // await screen.findByRole('combobox', { name: /community service delivery point/i });
```
Comment (samuelmale) [inline src/form-engine.test.tsx:119]: @denniskigen was this an oversight?

**Review by samuelmale — APPROVED** [review]

File: src/form-engine.test.tsx (hunk 1 above)
Comment (pirupius) [inline src/form-engine.test.tsx:119]: Was this intentional or don't we need them anymore?

**Review by pirupius — COMMENTED** [review]
LGTM. Just the commented code

File: src/form-engine.test.tsx (hunk 1 above)
Comment (denniskigen) [reply src/form-engine.test.tsx]: Fixed

**Pull request merged**